src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/DefaultJavaScriptEncoderBasicLatin.cs

```diff
@@ -166,9 +178,21 @@ public override unsafe int FindFirstCharacterToEncode(char* text, int textLength
 
                     // Load the next 8 characters + a dummy known that it must not be escaped.
                     // Put the dummy second, so it's easier for GetIndexOfFirstNeedToEscape.
-                    Vector128<sbyte> sourceValue = Sse2.PackSignedSaturate(
-                        Sse2.LoadVector128(ptr),
-                        Vector128.Create((short)'A'));  // max. one "iteration", so no need to cache this vector
+                    Vector128<sbyte> sourceValue;
+
+                    if (Sse2.IsSupported)
+                    {
+                        sourceValue = Sse2.PackSignedSaturate(
+                            Sse2.LoadVector128(ptr),
+                            Vector128.Create((short)'A'));  // max. one "iteration", so no need to cache this vector
+                    }
+                    else
+                    {
+                        Debug.Assert(AdvSimd.Arm64.IsSupported);
+
+                        Vector64<sbyte> saturated = AdvSimd.ExtractNarrowingSaturateLower(AdvSimd.LoadVector128(ptr));
+                        sourceValue = Vector128.Create(saturated, Vector64.Create((sbyte)'A'));
```

Comment:
>Vector64.Create((sbyte)'A') [](start = 66, length = 27)

nit: you can have a local variable that creates this and use it for both SSE2/AdvSimd case.

Reply:
They have different types, the SSE2 vector is of type short.

Reply:
Ah, i see.

---
In reply to: [453072121](https://github.com/dotnet/runtime/pull/38707#discussion_r453072121) [](ancestors = 453072121)